connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -540,10 +601,13 @@ Map<String, Config> describeTopicConfigs(Set<String> topics)
 
     Config targetConfig(Config sourceConfig) {
         List<ConfigEntry> entries = sourceConfig.entries().stream()
-            .filter(x -> !x.isDefault() && !x.isReadOnly() && !x.isSensitive())
-            .filter(x -> x.source() != ConfigEntry.ConfigSource.STATIC_BROKER_CONFIG)
-            .filter(x -> shouldReplicateTopicConfigurationProperty(x.name()))
-            .collect(Collectors.toList());
+                    .filter(x -> (!useIncrementalAlterConfigs.equals(MirrorSourceConfig.NEVER_USE_INCREMENTAL_ALTER_CONFIG)
+                            || (x.isDefault() && shouldReplicateSourceDefault(x.name())))
+                            || (!x.isDefault() && useIncrementalAlterConfigs.equals(MirrorSourceConfig.NEVER_USE_INCREMENTAL_ALTER_CONFIG)))
+                    .filter(x -> !x.isReadOnly() && !x.isSensitive())
+                    .filter(x -> x.source() != ConfigEntry.ConfigSource.STATIC_BROKER_CONFIG)
+                    .filter(x -> shouldReplicateTopicConfigurationProperty(x.name()))
```

Comment:
This double-checks default values, which I don't believe is correct (we should rely solely on `ConfigPropertyFilter::shouldReplicateSourceDefault` for those).

Perhaps all of the logic can be moved to the `shouldReplicateTopicConfigurationProperty` method, including determining whether the property is a default or not (and how to handle that accordingly)?

Reply:
Hmm... I was imagining something slightly different:

For the `ConfigPropertyFilter` interface, the same property would never be tested against both methods (`shouldReplicateSourceDefault` and `shouldReplicateConfigProperty`). This lines up with the language in the KIP, which is that users "will be able to configure this method to return true in order to include source cluster's default configurations in the replication" (it mentions nothing about also having to return true from `shouldReplicateConfigProperty`). It also avoids some redundancy (you shouldn't have to change multiple methods to accommodate a single property).

Reply:
I now understand what you are saying. I have attempted to address the suggestion, please let me if it doesn't look right.